### PR TITLE
serializers: use control_number instead of BAI on facet_author_name

### PR DESCRIPTION
**Pull request by monaawi**
* INSPIR-2752

### CODE REVIEW

File: backend/inspirehep/records/marshmallow/utils.py [hunk 1]
```diff
@@ -15,14 +15,15 @@ def get_display_name_for_author_name(author_name):
 
 
 def get_facet_author_name_for_author(author):
-    author_ids = author.get("ids", [])
-    author_bai = get_values_for_schema(author_ids, "INSPIRE BAI")
-    bai = author_bai[0] if author_bai else "BAI"
+    control_number = (
+        author["control_number"] if author.get("control_number") else "NOREC"
```
Comment (michamos) [inline backend/inspirehep/records/marshmallow/utils.py]: There should always be a control number here. Maybe log an error if that's not the case and don't put it in the facet?

File: backend/inspirehep/records/marshmallow/utils.py [hunk 2]
```diff
@@ -15,14 +15,15 @@ def get_display_name_for_author_name(author_name):
...
+    )
     author_preferred_name = get_value(author, "name.preferred_name")
 
     if author_preferred_name:
-        return "{}_{}".format(bai, author_preferred_name)
+        return "{}_{}".format(control_number, author_preferred_name)
```
Comment (michamos) [inline backend/inspirehep/records/marshmallow/utils.py]: Use f-strings instead of `format` (wherever it makes sense):
```suggestion
        return f"{control_number}_{author_preferred_name}"
```

File: backend/tests/integration/disambiguation/test_tasks.py [hunk 3]
```diff
@@ -84,6 +85,43 @@ def test_disambiguate_signatures_cluster_with_0_authors(
     assert expected_ref == record["authors"][0]["record"]["$ref"]
 
 
+@freeze_time("2019-02-15")
```
Comment (michamos) [inline backend/tests/integration/disambiguation/test_tasks.py]: I don't see any assertions on dates. Why is this needed?

File: backend/tests/integration/disambiguation/test_tasks.py [hunk 4]
```diff
@@ -84,6 +85,43 @@ def test_disambiguate_signatures_cluster_with_0_authors(
...
+def test_disambiguate_signatures_cluster_creates_author_with_facet_author_name(
```
Comment (michamos) [inline backend/tests/integration/disambiguation/test_tasks.py]: how is this different from the async test you also added?

File: backend/inspirehep/records/marshmallow/utils.py (hunk 1 above)
Comment (monaawi) [reply backend/inspirehep/records/marshmallow/utils.py]: but in the schema the control number isn't required, this is why I added this. Do you think this check should be skipped?
What do you mean by don't put it in the facet? Put NOREC instead? I am already doing this.

File: backend/tests/integration/disambiguation/test_tasks.py (hunk 4 above)
Comment (monaawi) [reply backend/tests/integration/disambiguation/test_tasks.py]: In the async one I'm checking the facet_author_name on the literature record once an author is linked by disambiguation (I moved it to async so I don't trigger the indexing myself, can also be moved to the integration tests and I would trigger the indexing, wdyt?), I guess the naming is misleading. In the integration one, I'm checking the facet_author_name on a newly created author (from the author's detail page)

File: backend/inspirehep/records/marshmallow/utils.py (hunk 1 above)
Comment (monaawi) [reply backend/inspirehep/records/marshmallow/utils.py]: Actually, I don't think that this check is necessary anymore - I think I was being paranoid, I'll remove it. This method is either called in the literature serializer (where we are already retrieving linked authors by id so it has to be there) or from the author details serializer.

Comment (michamos) [reply backend/inspirehep/records/marshmallow/utils.py]: We had some records in the past without a control number. It shouldn't be happening, so I think it's OK to blow up.

File: backend/tests/integration-async/disambiguation/test_disambiguation_tasks.py [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+from invenio_db import db
+from invenio_pidstore.models import PersistentIdentifier
+from invenio_search import current_search_client as es
+
+from inspirehep.disambiguation.tasks import disambiguate_signatures
+from inspirehep.records.api import AuthorsRecord
+from inspirehep.records.api.literature import LiteratureRecord
+
+
+def test_author_created_by_disambiguation_has_author_facet_name(
```
Comment (michamos) [inline backend/tests/integration-async/disambiguation/test_disambiguation_tasks.py]: then maybe
```suggestion
def test_signature_linked_by_disambiguation_has_correct_facet_author_name(
```
